vg/vgimg/vgimg.go

```diff
@@ -22,63 +22,138 @@ import (
 	"golang.org/x/image/tiff"
 )
 
-// dpi is the number of dots per inch.
-const dpi = 96
-
 // Canvas implements the vg.Canvas interface,
 // drawing to an image.Image using draw2d.
 type Canvas struct {
 	gc    draw2d.GraphicContext
-	img   image.Image
+	img   draw.Image
 	w, h  vg.Length
 	color []color.Color
 
+	// dpi is the number of dots per inch for this canvas.
+	dpi int
+
 	// width is the current line width.
 	width vg.Length
 }
 
-// New returns a new image canvas with
-// the size specified  rounded up to the
-// nearest pixel.
-func New(width, height vg.Length) *Canvas {
-	w := width / vg.Inch * dpi
-	h := height / vg.Inch * dpi
-	img := image.NewRGBA(image.Rect(0, 0, int(w), int(h)))
+// These constants are used to ensure that the options
+// used when initializing a canvas are compatible with
+// each other.
+const (
+	setsDPI = 1 << iota
+	setsSize
+
+	setsDPIAndSize = setsDPI | setsSize
```

Comment:
I would just or these together in the one place where they are used. No need to introduce a new name for it.